pkg/logger/logger.go

```diff
@@ -23,64 +23,64 @@ type Logger struct {
 	config := zap.NewProductionEncoderConfig()
 	config.EncodeTime = zapcore.ISO8601TimeEncoder
 	jsonEncoder := zapcore.NewJSONEncoder(config)
 
-	defaultLogLevel := zapcore.DebugLevel
-
-	core := zapcore.NewTee(zapcore.NewCore(jsonEncoder, zapcore.AddSync(os.Stdout), defaultLogLevel))
+	core := zapcore.NewTee(zapcore.NewCore(jsonEncoder, zapcore.AddSync(os.Stdout), logLevel))
 
 	zapLogger := zap.New(core, zap.AddCaller(), zap.AddStacktrace(zapcore.ErrorLevel))
 
 	return Logger{
 		zapLogger:     zapLogger,
-		requestId:     requestId,
 		errorReporter: errorReporter,
 	}
 }
 
-func (l *Logger) LogError(message string) {
-	l.zapLogger.Error(message, zap.String("requestId", l.requestId))
+func (l *Logger) SetRequestId(requestId string) {
+	l.requestId = requestId
 }
 
-func (l *Logger) LogAndReportUnexpectedError(message string) {
-	l.LogError(message)
-	l.errorReporter.ReportUnexpectedError(errors.New(message))
+func (l *Logger) LogDebug(message string, data ...any) {
+	l.zapLogger.Debug(message, zap.String("requestId", l.requestId), zap.Any("data", data))
+}
+
+func (l *Logger) LogInfo(message string, data ...any) {
+	l.zapLogger.Info(message, zap.String("requestId", l.requestId), zap.Any("data", data))
 }
 
-func (l *Logger) LogIncoming(admissionReview *admission.AdmissionReview) {
-	l.logInfo(admissionReview, "incoming")
+func (l *Logger) LogWarn(message string, data ...any) {
+	l.zapLogger.Warn(message, zap.String("requestId", l.requestId), zap.Any("data", data))
 }
-func (l *Logger) LogOutgoing(admissionReview *admission.AdmissionReview, isSkipped bool) {
-	l.logInfo(outgoingLog{
-		AdmissionReview: admissionReview,
-		IsSkipped:       isSkipped,
-	}, "outgoing")
+
+func (l *Logger) LogError(message string, data ...any) {
+	l.zapLogger.Error(message, zap.String("requestId", l.requestId), zap.Any("data", data))
 }
 
-type outgoingLog struct {
-	AdmissionReview *admission.AdmissionReview
-	IsSkipped       bool
+func (l *Logger) Fatal(message string, data ...any) {
+	l.zapLogger.Fatal(message, zap.String("requestId", l.requestId), zap.Any("data", data))
 }
 
-func (l *Logger) LogInfo(objectToLog any) {
-	l.logInfo(objectToLog, "")
+func (l *Logger) PanicLevel(message string, data ...any) {
+	l.zapLogger.Panic(message, zap.String("requestId", l.requestId), zap.Any("data", data))
 }
 
-// LogUtil this method creates a new logger instance on every call, and does not have a requestId
-// please prefer using the logger instance from the context instead
-func LogUtil(msg string) {
-	logger := New("", nil)
-	logger.LogInfo(msg)
+func (l *Logger) LogAndReportUnexpectedError(message string) {
+	l.LogError(message)
+	l.errorReporter.ReportUnexpectedError(errors.New(message))
 }
 
-func (l *Logger) logInfo(objectToLog any, requestDirection string) {
+func (l *Logger) LogAdmissionRequest(admissionReview *admission.AdmissionReview, isSkipped bool, direction string) {
+	if direction != "incoming" && direction != "outgoing" {
```

Comment:
This is not perfect but I think it's better
![image](https://github.com/datreeio/admission-webhook-datree/assets/39004075/597b90bd-4956-4a61-b861-daaac70955cc)